sympy/sets/sets.py

```diff
@@ -1738,23 +1738,25 @@ def _intersect(self, other):
         return self.__class__(el for el in self if el in other)
 
     def _complement(self, other):
-        if other is S.Reals:
+        if isinstance(other, Interval):
             nums = sorted(m for m in self.args if m.is_number)
-            syms = [m for m in self.args if m.is_Symbol]
-            # Reals cannot contain elements other than numbers and symbols.
+            if other is S.Reals and nums != []:
```

Comment:
That's because `S.Reals` derives from `Interval` class, See [this](https://github.com/sympy/sympy/blob/master/sympy/sets/fancysets.py#L175).

For Example `S.Complex` derives from `ComplexPlane` , and
`ComplexPlane` derives from `Set` so:

```python
In [15]: isinstance(S.Complex, ComplexPlane)
Out[15]: True

In [16]: isinstance(S.Complex, Set)
Out[16]: True
```

But in this case: `S.Reals` derives from `Interval` but S.Reals `is not` Interval.
Also `S.Reals` is `Interval(-oo, oo)` is also `False`, but
`S.Reals == Interval(-oo, oo)` return `True` because it has been made to do so [here](https://github.com/sympy/sympy/blob/master/sympy/sets/fancysets.py#L180).

Note: `is` and `==` are not same. `is` will return `True` if two variables point to the same object, and `==` if the objects referred to by the variables are equal.